src/main/java/gyro/aws/secretsmanager/SecretResource.java

```diff
@@ -0,0 +1,456 @@
+     */
+    @ConflictsWith("recovery-window-in-days")
+    public Boolean getForceDeleteWithoutRecovery() {
+        return forceDeleteWithoutRecovery;
+    }
+
+    public void setForceDeleteWithoutRecovery(Boolean forceDeleteWithoutRecovery) {
+        this.forceDeleteWithoutRecovery = forceDeleteWithoutRecovery;
+    }
+
+    /**
+     * The most recent date and time that this secret was accessed.
+     */
+    @Output
+    public String getLastAccessedDate() {
+        return lastAccessedDate;
+    }
+
+    public void setLastAccessedDate(String lastAccessedDate) {
+        this.lastAccessedDate = lastAccessedDate;
+    }
+
+    /**
+     * The most recent date and time that this secret was modified in any way.
+     */
+    @Output
+    public String getLastChangedDate() {
+        return lastChangedDate;
+    }
+
+    public void setLastChangedDate(String lastChangedDate) {
+        this.lastChangedDate = lastChangedDate;
+    }
+
+    /**
+     * The most recent date and time that the Secrets Manager rotation process was successfully completed.
+     */
+    @Output
+    public String getLastRotatedDate() {
+        return lastRotatedDate;
+    }
+
+    public void setLastRotatedDate(String lastRotatedDate) {
+        this.lastRotatedDate = lastRotatedDate;
+    }
+
+    /**
+     * The name of the secret. (Required)
+     */
+    @Required
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The name of the owning service.
+     */
+    @Output
+    public String getOwningService() {
+        return owningService;
+    }
+
+    public void setOwningService(String owningService) {
+        this.owningService = owningService;
+    }
+
+    /**
+     * The number of days that Secrets Manager waits before it can delete the secret.
+     */
+    @ConflictsWith("force-delete-without-recovery")
+    public Long getRecoveryWindowInDays() {
+        return recoveryWindowInDays;
+    }
+
+    public void setRecoveryWindowInDays(Long recoveryWindowInDays) {
+        this.recoveryWindowInDays = recoveryWindowInDays;
+    }
+
+    /**
+     * Enable automatic rotation for the secret.
+     */
+    @Output
+    public Boolean getRotationEnabled() {
+        return rotationEnabled;
+    }
+
+    public void setRotationEnabled(Boolean rotationEnabled) {
+        this.rotationEnabled = rotationEnabled;
+    }
+
+    /**
+     * The unique ARN of a Lambda function that's invoked by Secrets Manager to rotate the secret either automatically
+     * per the schedule or manually by a call to RotateSecret.
+     */
+    @Output
+    public String getRotationLambdaARN() {
+        return rotationLambdaARN;
+    }
+
+    public void setRotationLambdaARN(String rotationLambdaARN) {
+        this.rotationLambdaARN = rotationLambdaARN;
+    }
+
+    /**
+     * The structure that contains the rotation configuration for the secret.
+     */
```

Comment:
This helps generate documentation for the subresource

```suggestion
    /**
     * The structure that contains the rotation configuration for the secret.
     *
     * @subresource gyro.aws.secretsmanager.SecretRotationRulesType
     */
```